app/assets/javascripts/organization-filtering.js

```diff
@@ -0,0 +1,36 @@
+$(document).ready(function() {
+  const $searchAndSort = $("#js-search-and-sort-component");
+
+  if (!$searchAndSort) return;
+
+  const debounce = (function() {
+    let timer = 0;
+
+    return function(callback, ms) {
+      if (timer) {
+        clearTimeout(timer);
+      }
+      timer = setTimeout(callback, ms);
+    };
+  })();
+
+  $("#js-filtering-component").on("change keyup input", function(e) {
+    const componentContext = $(e.target).closest("#js-filtering-component");
+    const $searchForm = componentContext.find("#js-filtering-form");
+    const formData = $searchForm.find('input[name!=utf8]').serialize();
+
+    history.replaceState(null, '', '?' + formData);
+    debounce(function() { $searchForm.submit(); }, 300);
+  });
+
+  $("#js-filtering-component .SelectMenu-item").on("change", function(e) {
+    const $clickedItem = $(e.target).closest(".SelectMenu-item");
+    const $currentActiveItem = $clickedItem.siblings("[aria-checked='true']");
```

Comment:
Relying on `aria-checked` feels new to me. Is this a pattern we're using elsewhere?

Reply:
In a previous iteration where we rolled our own dropdown menus, we were relying on the addition of a `selected` class to indicate which item was selected. With that, we were looking for that class to determine which item had previously been selected.

With the Primer dropdown component, the item is selected using `aria-checked=true`. I was keeping the spirit of the code (finding the previously selected item by looking for the markup that indicated the item was selected) but updating it to reflect the way the Primer component worked.

I've updated to add the `selected` class so we can search based on the class instead. This would be consistent with what we were doing before; however, it means we now have to update both the class and the `aria-checked` attribute. It's not a large amount of extra code, but it does feel a bit `extra`. Is it a best practice in JS no to select using aria attributes?

Reply:
No not necessarily. Your reasoning makes perfect sense (thank you!) - based on that I would say let's use only the aria-checked like you originally had.